src/HotChocolate/Core/src/Execution/DependencyInjection/RequestExecutorBuilderExtensions.ErrorFilter.cs

```diff
@@ -57,7 +57,10 @@ public static partial class RequestExecutorBuilderExtensions
         }
 
         builder.Services.TryAddTransient<T>();
-        return builder.ConfigureSchemaServices(s => s.AddSingleton<IErrorFilter, T>());
+        return builder.ConfigureSchemaServices(s
+            => s.AddSingleton<T>()
```

Comment:
This still needs to be fixed.

Reply:
was this `sp.GetApplicationService()`?

Reply:
The error here was that you registered T twice ... inside the schema di and on the outer app DI. I have reworked that now to register T on the outer DI and getting it from there into the inner DI